txnkv/transaction/txn.go

```diff
@@ -789,6 +792,23 @@ func (txn *KVTxn) LockKeys(ctx context.Context, lockCtx *tikv.LockCtx, keysInput
 	return nil
 }
 
+// unsetPrimaryKeyIfNeed is used to unset pk of current transaction.
+// When lock only one key with LockOnlyIfExists flag, this key is selected as pk, if the key doesn't
+// exist on TiKV, it doesn't generate a lock for this key, we should unset this pk for this scene.
+// The caller should ensure bellow conditions.
+// (1) only one key to be locked (2) pk is not selected before (2) with LockOnlyIfExists
+// retun true means the pk has been unseted
```

Comment:
```suggestion
// unsetPrimaryKeyIfNeed is used to unset pk of the transaction after performing LockOnlyIfExists.
// When locking only one key with LockOnlyIfExists flag, the key will be selected as primary if
// it's the first lock of the transaction. If the key doesn't exist on TiKV, the key won't be
// locked, in which case we should unset the primary of the transaction.
// The caller must ensure bellow conditions:
// (1) only one key to be locked (2) primary is not selected before (2) with LockOnlyIfExists
// Returns true if the primary has been unset.
```

&ast;Trying to make the English expression more clear...

* Mention the requirement of LockOnlyIfExists in the first sentence to avoid potential misusing
* Prefer not using `pk` to stand for the primary of a transaction in documentation (I think)
* Some grammar adjustments...